### ISSUE TITLE
2.0.0a1 Closing bracket not forced into new line

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#4233]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm not entirely sure if this is within scope of SQLFluff 2.0 but as a user I would hope to allow as little ambiguity as possible.

For example the following SQL

```sql
SELECT *
FROM (
    SELECT *
    FROM table)
```

has the closing bracket not on a new line. But SQLFluff doesn't throw any lint warnings, and hence allows 2 ways of writing this SQL.

Intuitively I would have expected that 

```sql
SELECT *
FROM (
    SELECT *
    FROM table
)
```

Is the only "correct" way of writing this query.

### Expected Behaviour

```sql
SELECT *
FROM (
    SELECT *
    FROM table
)
```

### Observed Behaviour

No lint issues

### How to reproduce

Run `sqlfluff lint` on

```sql
SELECT *
FROM (
    SELECT *
    FROM table)
```

### Dialect

ansi

### Version

2.0.0a1

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Agreed - this is actually a duplicate of #4222 and was fixed in #4236
